### PR TITLE
feat(API): Moving properties to relevant classes

### PR DESCRIPTION
**Pull request by jashnani**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst).
- [x] New tests have been created for any new features or regression tests for bugfixes.
- [x] `tox` successfully runs, including unit tests and style checks (see [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst)).

### What does this Pull Request accomplish?

Moves properties to relevant sub classes. 

### Why should this Pull Request be merged?

Prevents the user from using the attribute with unsupported session classes (that could otherwise lead to errors or do nothing and leave dead code behind). 

### What testing has been done?

None

### CODE REVIEW

**Comment by jashnani** [comment]
Added issue #121 to track that one of these needs to be moved to a class that doesn't exist yet.

File: nixnet/_session/frames.py [hunk 1]
```diff
@@ -22,6 +22,23 @@ def __repr__(self):
     def _create_item(self, handle, index, name):
         return Frame(handle, index, name)
 
+    @property
+    def payld_len_max(self):
+        """int: Returns the maximum payload length of all frames in this session, expressed as bytes (0-254).
+
+        For CAN Stream (Input and Output), this property depends on the XNET
+        Cluster CAN I/O Mode property. If the I/O mode is CAN, this property is
```
Comment (epage) [inline nixnet/_session/frames.py]: Looks inconsistent in referring to the enum values, so `CAN` -> `constants.CaNioMode.CAN`?

File: nixnet/_session/frames.py [hunk 2]
```diff
@@ -22,6 +22,23 @@ def __repr__(self):
...
+        8 bytes. If the I/O mode is 'constants.CaNioMode.CANFD' or
+        'constants.CaNioMode.CANFD', this property is 64 bytes.
+
+        For LIN Stream (Input and Output), this property always is 8 bytes. For
+        FlexRay Stream (Input and Output), this property is the same as the XNET
+        Cluster FlexRay Payload Length Maximum property value. For Queued and
+        Single-Point (Input and Output), this is the maximum payload of all
+        frames specified in the List property.
```
Comment (epage) [inline nixnet/_session/frames.py]: The comment about Queued and Single-point, is that applying to all protocols?

If so, that should be a separate paragraph.

(the fun of drawing attention to previous work)

File: nixnet/_session/signals.py [hunk 3]
```diff
@@ -17,6 +18,18 @@ def __repr__(self):
     def _create_item(self, handle, index, name):
         return Signal(handle, index, name)
 
+    @property
+    def resamp_rate(self):
+        """float: Rate used to resample frame data to/from signal data in waveforms.
+
+        The data type is 64-bit floating point. The units are in Hertz (samples per second).
```
Comment (epage) [inline nixnet/_session/signals.py]: > he data type is 64-bit floating point

Thats already conveyed with the `float:`

File: nixnet/_session/frames.py (hunk 1 above)
Comment (jashnani) [inline nixnet/_session/frames.py]: Fixed.

File: nixnet/_session/frames.py (hunk 2 above)
Comment (jashnani) [inline nixnet/_session/frames.py]: Moved to separate para. ditto for FlexRay.

File: nixnet/_session/signals.py (hunk 3 above)
Comment (jashnani) [inline nixnet/_session/signals.py]: Removed comment.

File: nixnet/_session/frames.py [hunk 4]
```diff
@@ -22,6 +22,25 @@ def __repr__(self):
     def _create_item(self, handle, index, name):
         return Frame(handle, index, name)
 
+    @property
+    def payld_len_max(self):
```
Comment (epage) [inline nixnet/_session/frames.py]: How did this move not break our tests?

https://github.com/ni/nixnet-python/blob/master/tests/test_frames.py#L216

File: nixnet/_session/signals.py [hunk 5]
```diff
@@ -17,6 +18,18 @@ def __repr__(self):
     def _create_item(self, handle, index, name):
         return Signal(handle, index, name)
 
+    @property
+    def resamp_rate(self):
```
Comment (epage) [inline nixnet/_session/signals.py]: This should have also broken the tests
https://github.com/ni/nixnet-python/blob/master/tests/test_frames.py#L222

File: nixnet/_session/frames.py (hunk 4 above)
Comment (jashnani) [inline nixnet/_session/frames.py]: I think this has to do with the Jenkins CI not running everytime a PR is posted or even sometimes when a PR is merged. @lilesj do you know why the CI didn't run this time?

Comment (lilesj) [inline nixnet/_session/frames.py]: I am not sure why it didn't run. The current Jenkinsfile in the master branch only clones the repo and doesn't actually perform any tests.

Comment (epage) [inline nixnet/_session/frames.py]: Oh, right, thats in an integration test.  Oops.

Comment (jashnani) [inline nixnet/_session/frames.py]: I've fixed test for now.

File: nixnet/_session/signals.py (hunk 5 above)
Comment (jashnani) [inline nixnet/_session/signals.py]: Ditto.

**Pull request merged**